diff --git a/splatnet2statink.py b/splatnet2statink.py
index 6edab78..5c9d123 100755
--- a/splatnet2statink.py
+++ b/splatnet2statink.py
@@ -1219,7 +1219,7 @@ def post_battle(i, results, s_flag, t_flag, m_flag, sendgears, debug, ismonitor=
 		url  = 'https://stat.ink/api/v2/battle'
 		auth = {'Authorization': 'Bearer {}'.format(API_KEY), 'Content-Type': 'application/x-msgpack'}
 
-		if payload["agent"] != os.path.basename(__file__)[:-3]:
+		if payload["agent"] != os.path.splitext(os.path.basename(__file__))[0]:
 			print("Could not upload. Please contact @frozenpandaman on Twitter/GitHub for assistance.")
 			sys.exit(1)
 		postbattle = requests.post(url, headers=auth, data=msgpack.packb(payload), allow_redirects=False)
