diff --git a/src/abomination/meson.build b/src/abomination/meson.build
index 7f4dbc7..f3e78e8 100644
--- a/src/abomination/meson.build
+++ b/src/abomination/meson.build
@@ -8,6 +8,7 @@ libabomination_deps = [
     dep_gtk3,
     dep_gdkx11,
     dep_wnck,
+    dep_giounix,
 	link_libappsys,
 ]
 
diff --git a/src/applets/budgie-menu/meson.build b/src/applets/budgie-menu/meson.build
index 79d7b00..c3a75fb 100644
--- a/src/applets/budgie-menu/meson.build
+++ b/src/applets/budgie-menu/meson.build
@@ -31,6 +31,7 @@ applet_budgiemenu_deps = [
     libplugin_vapi,
     dep_gtk3,
     dep_peas,
+    dep_giounix,
     link_libplugin,
     dependency('libgnome-menu-3.0', version: '>= 3.10.3'),
 ]
diff --git a/src/applets/clock/meson.build b/src/applets/clock/meson.build
index 537280c..790ef03 100644
--- a/src/applets/clock/meson.build
+++ b/src/applets/clock/meson.build
@@ -17,6 +17,7 @@ applet_clock_deps = [
     libplugin_vapi,
     dep_gtk3,
     dep_peas,
+    dep_giounix,
     link_libplugin,
 ]
 
diff --git a/src/applets/icon-tasklist/meson.build b/src/applets/icon-tasklist/meson.build
index 6ba20c9..8d67f98 100644
--- a/src/applets/icon-tasklist/meson.build
+++ b/src/applets/icon-tasklist/meson.build
@@ -36,6 +36,7 @@ applet_icontasklist_deps = [
     dep_peas,
     dep_wnck,
     dep_gdkx11,
+    dep_giounix,
     link_libabomination,
     link_libappsys,
     link_libplugin,
diff --git a/src/applets/night-light/meson.build b/src/applets/night-light/meson.build
index a5f8836..053f8ae 100644
--- a/src/applets/night-light/meson.build
+++ b/src/applets/night-light/meson.build
@@ -29,6 +29,7 @@ applet_nightlight_deps = [
     libplugin_vapi,
     dep_gtk3,
     dep_peas,
+    dep_giounix,
     link_libplugin,
 ]
 
diff --git a/src/applets/status/meson.build b/src/applets/status/meson.build
index 54926ba..e14ce99 100644
--- a/src/applets/status/meson.build
+++ b/src/applets/status/meson.build
@@ -31,6 +31,7 @@ applet_status_deps = [
     dep_gtk3,
     dep_peas,
     dep_accountsservice,
+    dep_giounix,
     dependency('gnome-bluetooth-1.0', version: '>= 3.34.0'),
     dependency('upower-glib', version: '>= 0.99.0'),
     link_libplugin,
diff --git a/src/applets/workspaces/meson.build b/src/applets/workspaces/meson.build
index d931d3d..e8c4ffd 100644
--- a/src/applets/workspaces/meson.build
+++ b/src/applets/workspaces/meson.build
@@ -31,6 +31,7 @@ applet_workspaces_deps = [
     dep_gtk3,
     dep_wnck,
     dep_peas,
+    dep_giounix,
     link_libappsys,
     link_libplugin,
 ]
diff --git a/src/appsys/meson.build b/src/appsys/meson.build
index 14a92a5..3e3682a 100644
--- a/src/appsys/meson.build
+++ b/src/appsys/meson.build
@@ -11,6 +11,7 @@ libappsys = static_library(
         dep_gtk3,
         dep_gdkx11,
         dep_wnck,
+        dep_giounix,
         link_libconfig,
     ],
     vala_args: [
diff --git a/src/daemon/meson.build b/src/daemon/meson.build
index e78918b..304f87d 100644
--- a/src/daemon/meson.build
+++ b/src/daemon/meson.build
@@ -31,6 +31,7 @@ daemon_deps = [
     dep_gdkx11,
     dep_notify,
     dep_wnck,
+    dep_giounix,
     link_libappsys,
     link_libconfig,
     link_libtheme,
diff --git a/src/lib/meson.build b/src/lib/meson.build
index dbbd49b..e34dcbe 100644
--- a/src/lib/meson.build
+++ b/src/lib/meson.build
@@ -14,6 +14,7 @@ libbudgieprivate = shared_library(
         libplugin_vapi,
         dep_peas,
         dep_gtk3,
+        dep_giounix,
         meson.get_compiler('c').find_library('m', required: false),
         link_libplugin,
     ],
diff --git a/src/libsession/meson.build b/src/libsession/meson.build
index 9121d6b..24dd768 100644
--- a/src/libsession/meson.build
+++ b/src/libsession/meson.build
@@ -7,7 +7,10 @@ libsession_sources = [
 libsession = static_library(
     'session',
     libsession_sources,
-    dependencies: dep_glib,
+    dependencies: [
+        dep_glib,
+        dep_giounix,
+    ],
     vala_args: [
         '--pkg', 'gio-unix-2.0',
     ],
diff --git a/src/panel/meson.build b/src/panel/meson.build
index 5451c27..51ed5e1 100644
--- a/src/panel/meson.build
+++ b/src/panel/meson.build
@@ -45,6 +45,7 @@ panel_deps = [
     dep_peas,
     dep_libuuid,
     dep_wnck,
+    dep_giounix,
     link_libconfig,
     link_libsession,
     link_libbudgieprivate,
diff --git a/src/plugin/meson.build b/src/plugin/meson.build
index bb63c72..e589ee5 100644
--- a/src/plugin/meson.build
+++ b/src/plugin/meson.build
@@ -47,6 +47,7 @@ plugin_sources = [
 plugin_deps = [
     dep_gtk3,
     dep_peas,
+    dep_giounix,
 ]
 
 # build plugin
diff --git a/src/polkit/meson.build b/src/polkit/meson.build
index 559810f..13382af 100644
--- a/src/polkit/meson.build
+++ b/src/polkit/meson.build
@@ -25,6 +25,7 @@ pkversion = '>= 0.105'
 
 polkit_deps = [
     dep_gtk3,
+    dep_giounix,
     link_libconfig,
     link_libtheme,
     link_libsession,
diff --git a/src/raven/meson.build b/src/raven/meson.build
index 2830e39..517d486 100644
--- a/src/raven/meson.build
+++ b/src/raven/meson.build
@@ -37,6 +37,7 @@ libraven_deps = [
     libplugin_vapi,
     dep_gtk3,
     dep_peas,
+    dep_giounix,
     link_libconfig,
     link_libplugin,
     link_libtheme,
diff --git a/src/rundialog/meson.build b/src/rundialog/meson.build
index ab0b20b..abb0b1e 100644
--- a/src/rundialog/meson.build
+++ b/src/rundialog/meson.build
@@ -5,6 +5,7 @@ rundialog_sources = [
 
 rundialog_deps = [
     dep_gtk3,
+    dep_giounix,
     link_libconfig,
     link_libtheme,
 ]
diff --git a/src/wm/meson.build b/src/wm/meson.build
index 17e6bc0..6bcd54c 100644
--- a/src/wm/meson.build
+++ b/src/wm/meson.build
@@ -46,6 +46,7 @@ budgie_wm_deps = [
     dep_gnomedesktop,
     dep_gsettings,
     dep_ibus,
+    dep_giounix,
 ]
 
 if dep_gsd.version().version_compare('>=3.36.0')
